src/cmp_ccpa.js

```diff
@@ -132,28 +132,35 @@ const loadPage = async (page, url) => {
 const checkPage = async (url) => {
 	logInfoMessage(`Start checking Page URL: ${url}`);
 
-	const browser = await makeNewBrowser();
-	const page = await browser.newPage();
+	let browser = null;
+	try {
+		const browser = await makeNewBrowser();
+		const page = await browser.newPage();
 
-	// Clear cookies before starting testing, to ensure the CMP is displayed.
-	const client = await page.target().createCDPSession();
-	await clearCookies(client);
+		// Clear cookies before starting testing, to ensure the CMP is displayed.
+		const client = await page.target().createCDPSession();
+		await clearCookies(client);
 
-	await loadPage(page, url);
+		await loadPage(page, url);
 
-	await checkTopAdHasLoaded(page);
+		await checkTopAdHasLoaded(page);
 
-	await checkCMPIsOnPage(page);
+		await checkCMPIsOnPage(page);
 
-	await interactWithCMP(page);
+		await interactWithCMP(page);
 
-	await checkCMPIsNotVisible(page);
+		await checkCMPIsNotVisible(page);
 
-	await reloadPage(page);
+		await reloadPage(page);
 
-	await checkTopAdHasLoaded(page);
-
-	await browser.close();
+		await checkTopAdHasLoaded(page);
+	} catch (error) {
+		logErrorMessage(error);
+	} finally {
```

Comment:
Had to do a very similar thing in the CMP monitoring lambdas: https://github.com/guardian/consent-management-platform/pull/620/files#diff-be624d3d93e2d5cc2356d1abb2e5faa442603342d873040b4a2c041816e8194dR131

Reply:
That's where I got the idea from 😄